harness/ts/harness.ts

```diff
@@ -185,4 +193,36 @@ export class Runner<W extends Workflow, A extends ActivityInterface> {
     await this.workerRunPromise;
     await this.nativeConnection.close();
   }
+
+  /**
+   * Temporary workaround for the buggy implmentation of this method in the SDK.
+   * TODO(bergundy): remove this when SDK issue is fixed (https://github.com/temporalio/sdk-typescript/issues/773)
+   */
+  payloadToJSON({ metadata, data }: Payload): JSONPayload {
+    return {
+      metadata:
+        metadata &&
+        Object.fromEntries(
+          Object.entries(metadata).map(([k, v]): [string, string] => [k, Buffer.from(v).toString('base64')])
+        ),
+      data: data ? Buffer.from(data).toString('base64') : undefined,
+    };
+  }
+
+  async getHistoryEvents(handle: WorkflowHandle): Promise<proto.temporal.api.history.v1.IHistoryEvent[]> {
+    let nextPageToken: Uint8Array | undefined = undefined;
+    const history = Array<proto.temporal.api.history.v1.IHistoryEvent>();
+    for (;;) {
+      const response: proto.temporal.api.workflowservice.v1.GetWorkflowExecutionHistoryResponse =
```

Comment:
Does this need to be explicitly typed?

Reply:
Unfortunately yes, I'm not sure what the reason was but adding an explicit type fixed the compiler error I was seeing.
It's been a while now so I'd have to see.